ssm/invocation/types.go

```diff
@@ -15,16 +15,39 @@ type CommandOutputSafe struct {
 	Output []*ssm.SendCommandOutput
 }
 
-// Result is used to store information about an invocation run on a particular instance
+// SuccessResult is used to store information about an invocation run on a particular instance
 type Result struct {
 	InvocationResult *ssm.GetCommandInvocationOutput
 	ProfileName      string
 	Region           string
 	Status           string
 }
 
+// Error is used to store information about an error when attempting to run an invocation on a particular instance
+type Error struct {
```

Comment:
Rather then introducing a new type and a new array we need to loop over, lets flush out what a result is and add error types as a valid result. I think we can simply update the Result struct to have an error field and expand on the status to support our needs. Currently status is a set based on the response from SSM API. We should enumerate those values and introduce an error status for client errors to capture this. We can use this enumerated status to appropriately handle any client errors that can occur while attempting to run a command.

```go

// Status represents the status of run command
type Status string

const(
        CommandSuccess Status = "Success"
        CommandError Status = "Error"
        ClientError Status = "ClientError"
)

type Result struct {
	InvocationResult *ssm.GetCommandInvocationOutput
	ProfileName      string
	Region           string
	Status           Status

	Error            error
}
```

This results in only one array we need to iterate over and should keep ordering of the output. It's important to note that not all of the error captured with the current changes are 'pre invocation errors'. This captures the following error cases that each involve a different SSM API operation:

* SendCommand Failure - this is a pre invocation error
* ListCommands Failure - Performed after a successful SendCommand
* ListCommandInvocationsPages Failure - Attempting to list invocations with matching command ID
* GetCommandInvocation Failure - Attempting to get each individual result from every instance the command was ran against

Each of the above errors can result due to an incorrectly configured/invalid session. Those API calls can also fail for other reason such as a true 500 or as a potential result of API rate limiting (I believe this is an error). Currently the output of the results is ordered by command. This will keep that order for scenarios where there is a client error on SendCommand but will result in a weird ordering output for the other scenarios.